### PR TITLE
mintlist-cli: Adding a bump comand that diffs mintlists and overrides

### PR DESCRIPTION
**Pull request by terranmoccasin**
Adding a `bump` command that diffs mintlists to update the package version.

### CODE REVIEW

File: packages/mintlist-cli/src/cmd/bump.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    return "minor";
+  }
+  if (hasAdded || diffOverrides(before, after)) {
+    return "patch";
+  }
+}
+
+function diffOverrides(before: string, after: string): boolean {
+  const beforeOverrides = MintlistFileUtil.fromString<MetadataOverrides>(
+    execSync(`git show ${before}:src/overrides.json`, {
```
Comment (yugure-orca) [inline packages/mintlist-cli/src/cmd/bump.ts:53]: I think the flexibility of the options of gen-tokenlist makes the tool a bit complicated. How about making it a rule that src/overrides.json must be created for this tool, and removing the overrides option from the gen-tokenlist command.

**Comment by terranmoccasin** [comment]
Hey @yugure-orca thanks for the suggestions. I've addressed them all and this is ready for another look!

File: packages/mintlist-cli/src/cmd/bump.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    .trim()
+    .split("\n")[0];
+  return flag === "1";
+}
+
+// Checks if a mintlist.json file was added or removed from the src/mintlists directory
+// Returns true if a mintlist.json file was added or removed, false otherwise
+function hasMintlistChanges(before: string, after: string) {
+  const beforeMintlists = toMintlistFilePaths(
+    execSync(`git ls-tree --name-only ${before} src/mintlists`, {
```
[the remark this answers is not in the document]
Comment (yugure-orca) [reply packages/mintlist-cli/src/cmd/bump.ts]: @terranmoccasin 
Oh, sorry.

My point is that this bump constrains mintlist to be under src/mintlists directory only, but add-mint/remove-mint/gen-tokenlist does not have that constraint. ./hoge.mintlist.json and ./src/fuga.mintlist.json also appear to be possible.

As with overrides.json, how about forcing the mintlist file to be placed under src/mintlists, or have bump detect the mintlist file for the entire directory.

Personally, I think it is reasonable to force users to use the following directory/file structure.

```
src/overrides.json
src/mintlists/x.mintlist.json
src/mintlists/y.mintlist.json
...
```

**Comment by yugure-orca** [comment]
@terranmoccasin 
Thanks for the update!
I additionally commented only on how to find mintlist files in hasMintlistChanges. 🙏
Simple usage by Orca itself should not be a problem, but the flexibility of the options creates a problem.
IMO, it would be less trouble and less inconvenient to constrain it than to free it.

**Review by yugure-orca — APPROVED** [review]
LGTM!

**Pull request merged**